src/components/ApiDelegationOverviewPage/ApiDelegationAccordion/ApiDelegationAccordion.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState } from 'react';
+import {
+  Accordion,
+  AccordionHeader,
+  AccordionContent,
+  Button,
+  ButtonVariant,
+  List,
+  BorderStyle,
+} from '@altinn/altinn-design-system';
+import cn from 'classnames';
+
+import { DeletableListItem } from './DeletableListItem';
+import classes from './ApiDelegationAccordion.module.css';
+
+export interface DelegationAccordionProps {
+  name: string;
+  organizations: Array<{
+    id: number;
+    name: string;
+    isSoftDelete: boolean;
+  }>;
+  setOrganizations: (
+    newArray: Array<{
+      id: number;
+      name: string;
+      isSoftDelete: boolean;
+    }>,
+  ) => void;
+}
+
+export const ApiDelegationAccordion = ({ name, organizations, setOrganizations }: DelegationAccordionProps) => {
+  const [open, setOpen] = useState(false);
+  const [isAllSoftDeleted, setAllSoftDeleted] = useState(() => {
+    for (const item of organizations) {
+      if (!item.isSoftDelete) {
+        return false;
+      }
+    }
+    return true;
+  });
+
+  const handleAccordionClick = () => {
+    setOpen(!open);
+  };
+
+  const toggleSoftDelete = (orgId: number) => {
+    const newArray = [...organizations];
```

Comment:
But if we are to make the naming more generic later (which I was considering doing here already actually), since this component doesn't just work for displaying and deleting lists of orgs, then this would have to be changed back.
Personally, I don't think you lose readability by suing newArray - it's a temp array, used to hold the new state, exactly like the name suggests. In fact, we may actually gain value by making it more generic imo.

Reply:
Well, in that case we make the naming generic again. The problem with newArray is that you have to go up to the variable to see what it is newArray is a list of when you look at the for-loop.

Reply:
I mean... it's always less than an inch away from something that explains it, so I think it's a bit redundant. But ok, I'll change it to newOrgArray then? Then the name will tell you that this is to become the new state of the org array.

Reply:
https://medium.com/@taohidulii/naming-your-variables-f9477ba002e9

![image](https://user-images.githubusercontent.com/31471142/199514790-264cdf59-fd66-4a0e-8940-d52c74a0daf9.png)

Reply:
Although that is an extreme example there's never a con of naming the variables better.

Reply:
I know. My argument was that the variable name, as it stood, was sufficiently clear. Because you should always weigh the pro's and con's of adding more letters to a name because longer names are more tedious both to read and to write which again will take away from the readability.

I've changed the name here though, but to newOrgArray 🙂
I chose array instead of list, because it is of type array, and I stuck to new instead of copy because it is only a copy in the very first line of its existence. As soon as we change it, it is no longer a copy. Besides it is to become the new state and so I think this name explains the reason for its existence more clearly.

Reply:
Yeah and my point was that newArray wasn't sufficiently clear because you had to look on the destrucured object to see what the array contained. The naming NewOrgArray is nice, my point was that it should contain org. Also I think readability is only negatively affected when the variable names consists of long compsited words. In this case, adding 3 letters to suddenly make it possible for people to understand what the variable consists of without having to look at the destructured object it was made of outweighs the cons of making the variableName 3 letters longer.